src/Controllers/ImageController.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+namespace JustBetter\GlideDirective\Controllers;
+
+use Closure;
+use Illuminate\Http\Request;
+use Illuminate\Routing\Controller;
+use Illuminate\Support\Facades\Storage;
+use Illuminate\Support\Str;
+use League\Glide\Server;
+use League\Glide\Signatures\Signature;
+use League\Glide\Signatures\SignatureException;
+use Statamic\Assets\Asset;
+use Statamic\Contracts\Assets\Asset as AssetContract;
+use Statamic\Facades\Asset as AssetFacade;
+use Statamic\Imaging\ImageGenerator;
+use Symfony\Component\HttpFoundation\BinaryFileResponse;
+
+class ImageController extends Controller
+{
+    protected ?AssetContract $asset;
+
+    protected array $params;
+
+    public function __construct(protected ImageGenerator $imageGenerator, protected Server $server) {}
+
+    public function getImageByPreset(Request $request, string $preset, string $fit, string $signature, string $file, string $format): BinaryFileResponse
+    {
+        /** @var ?Asset $asset */
+        $asset = AssetFacade::findByUrl(Str::start($file, '/'));
+        $this->asset = $asset;
```

Comment:
We can directly set `$this->asset` here, i think the typehint in the comment is not needed then too